library/alloc/src/str.rs

```diff
@@ -153,30 +156,33 @@ where
     // if the `len` calculation overflows, we'll panic
     // we would have run out of memory anyway and the rest of the function requires
     // the entire Vec pre-allocated for safety
-    let len = sep_len
+    let reserved_len = sep_len
         .checked_mul(iter.len())
         .and_then(|n| {
             slice.iter().map(|s| s.borrow().as_ref().len()).try_fold(n, usize::checked_add)
         })
         .expect("attempt to join into collection with len > usize::MAX");
 
     // crucial for safety
```

Comment:
It seems like the loop now handles this more defensively, and the switch to `debug_assert!` reflects that. Could you revise this comment accordingly?

Reply:
This change was a small unrelated fix. `assert!` was redundant even before this PR, and I thought that "critical for safety" comment was referring to `with_capacity()` call. That said, I'll see if I can improve the safety comments to provide better context :)